diff --git a/create.js b/create.js
index c700ef9725cabcaf285e3163d85ae57ce770370a..9904c3b8fd5ef1bea5e8cde477583e81366ae217 100644
--- a/create.js
+++ b/create.js
@@ -27,7 +27,7 @@ export default function create (modules, options={}) {
     const data = { }
     for (let i = 0, max = names.length; max > i; i++) {
       const name = names[i]
-      
+
       const isDirective = name.indexOf(directive) === 0
 
       if (isDirective) {
@@ -47,6 +47,12 @@ export default function create (modules, options={}) {
         }
 
       } else {
+        // support Snabbdom class module via class objects:
+        if (name === 'class' && input[name] && typeof input[name] === 'object') {
+          data.class = input[name]
+          continue
+        }
+
         // put all other attributes into `data.attrs`
         if (isBoolAttribute(name) && input[name] === 'false')
           input[name] = false
@@ -63,10 +69,10 @@ export default function create (modules, options={}) {
 
   // create the snabbdom + hyperx functions
   const patch = init(modules || [ ])
-  
+
   // create snabby function
   const snabby = hyperx(createElement, { comments: true, attrToProp: false })
-  
+
   // create yo-yo-like update function
   snabby.update = function update (dest, src) {
     return patch(dest, src)
